platform/CMakeLists.txt

```diff
@@ -1,10 +1,21 @@
-set(OPENSSL_USE_STATIC_LIBS TRUE)
-find_package(OpenSSL REQUIRED)
+# For static library builds, link against static library of OpenSSL
+if(NOT BUILD_SHARED_LIBS)
+    set( OPENSSL_USE_STATIC_LIBS TRUE )
+endif()
+find_package(OpenSSL)
+# Verify the minimum OpenSSL version required
+if( ${OPENSSL_FOUND} AND ( OPENSSL_VERSION MATCHES "0.9$" OR OPENSSL_VERSION MATCHES "1.0$" ) )
+    message( WARNING "OpenSSL 1.1.0 or later required: OpenSSL ${OPENSSL_VERSION} found." )
+endif()
```

Comment:
Is it better to have these checks on the uppermost level of `platform/`, or keep them on the lowermost level of `platform/posix/transport/` as they were before?

Reply:
I was contemplating this as well and decided it was better to have it in a single file. Because CMake makes everything global, I'm trying to keep all shared build dependencies in the same files to make things easier to manage. We ended up duplicating these checks in so many places. Also, seems like it should be possible to check if OpenSSL is available even though the wrapper isn't being included. What happens when there is no wrapper like in the case of pthreads? Open to your thoughts though.